pkgs/servers/samba/4.x.nix

```
@@ -60,15 +64,25 @@ stdenv.mkDerivation rec {
     ./4.x-fix-makeflags-parsing.patch
     # Backport, should be removed for version 4.14
     ./0001-lib-util-Standardize-use-of-st_-acm-time-ns.patch
+    # allow external asn1_compile and compile_et to support cross-compilation
+    (fetchpatch {
+      url = "https://git.buildroot.net/buildroot/plain/package/samba4/0002-build-find-pre-built-heimdal-build-tools-in-case-of-.patch?id=a2b344a34d042021d8ceaabfe371dd824b1cc440";
+      sha256 = "0smcm6xn1psiqjlc4v1kxzqjblmfss53cmcc09i4izlq3l835946";
+    })
   ];
 
   nativeBuildInputs = [
+    python3
+    wafHook
     pkg-config
     bison
     flex
     perl
     perl.pkgs.ParseYapp
     libxslt
+    glibc # rpcgen
+    buildPackages.stdenv.cc
+    heimdal
```

It might be hard to see why this is in nativBuildInputs otherwise:

```suggestion
    heimdal # compile_et
```